en/middleware/micrortps.md

```diff
@@ -0,0 +1,261 @@
+# RTPS/ROS2 Interface: PX4-FastRTPS Bridge
+
+The *PX4-FastRTPS Bridge* adds a Real Time Publish Subscribe (RTPS) interface to PX4, enabling the exchange of [uORB messages](../middleware/uorb.md) between PX4 components and (offboard) *FastRTPS* applications.
+
+> **Note** RTPS is the underlying protocol of the Object Management Group's (OMG) Data Distribution Service (DDS) standard. It aims to enable scalable, real-time, dependable, high-performance and interoperable data communication using the publish/subscribe pattern. *FastRTPS* is a very lightweight cross-platform implementation of the latest version of the RTPS protocol and a minimum DDS API.
+
+RTPS has been adopted as the middleware for the ROS2 (Robot Operating System). The bridge allows us to better integrate with ROS2, making it easy to share sensor values, commands, and other vehicle information.
+
+This topic describes the bridge architecture, how it is compiled, and how to write a simple FastRTSP application to subscribe to PX4 changes.
+
+
+## Architectural overview
+
+![basic example flow](../../assets/middleware/micrortps/basic_example_flow.png)
+
+The main elements of the architecture are the client and agent processes shown in the diagram above.
+
+- The *Client* is PX4 middleware daemon process that runs on the flight controller. It subscribes to uORB topics published by other PX4 components and sends any updates to the *Agent* (via a UART or UDP port). It also receives messages from the *Agent* and publishes them as uORB message on PX4.
```

Comment:
Thanks for this clarification @VicenteEprosima. I have not updated the text to state this because I believe that users don't need to know this level of detail, and what we have now is accurate: 

> "The *Agent* and *Client* are connected via a serial link (UART) or UDP network. The uORB information is [CDR serialized](https://en.wikipedia.org/wiki/Common_Data_Representation) for sending (*CDR serialization* provides a common format for exchanging serial data between different platforms)."

Let me know if you disagree, and I can talk more about the format.